test/plugin/base_test.rb

```diff
@@ -1015,6 +1015,67 @@ def test_http_request_skip_setting_latency
     end
   end
 
+  def test_log_with_trace
```

Comment:
Can we split the tests for `keep_trace_key` and `trace_key` into two separate unit tests?